### PR TITLE
Add semantic slots, starting with text fields

### PR DESCRIPTION
**Pull request by phkuo**
#### Pull request checklist

- [ ] Addresses an existing issue: #0000
- [ ] Include a change request file if publishing 
- [x] New feature, bugfix, or enhancement
  - [ ] Includes tests
- [ ] Documentation update

#### Description of changes

Lay some groundwork for converting to semantic slots, starting with input text fields.

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by micahgodbolt** [comment]
Looks fine to me (assuming no visual regressions).  I've seen this approach before from @phkuo and it closely mirrors how I'd do it with external Theo tokens. I'm mostly curious how this would look using Glamor, though I don't feel anything here would make converting to Glamor harder.

Maybe the bigger question is how close are we to converting to Glamor (pretty close I think), and should this work just be done in coordination with that glamor conversion.

**Comment by micahgodbolt** [comment]
@dzearing how are you feeling about this approach and what it'd look like with Glamor? I don't see any problem creating something similar, but wanted to get your feedback before we merge.

And @phkuo, just an FYI that you have some conflicts to resolve.

File: packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.scss [hunk 1]
```diff
@@ -3,6 +3,27 @@
 
 // Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See LICENSE in the project root for license information.
 
+
+// --------------------------------------------------
+// ChoiceField semantic slots
+
+$radioButton-background-color: $controlBackgroundColor;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.scss]: is a radio button any different frmo a checkbox?

Comment (phkuo) [inline packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.scss]: Unfortunately yes. For example, in the selected state, it uses a blue background and a white checkmark, whereas radio buttons have a white background and a blue dot.

**Comment by phkuo** [comment]
Doing design review before completing this PR.

**Comment by phkuo** [comment]
@dzearing @micahgodbolt Could I get this reviewed and signed off please?

**Comment by micahgodbolt** [comment]
@phkuo a few conflicts to resolve in textfield

**Pull request merged**